include/f_core/os/c_tenant.h

```diff
@@ -0,0 +1,41 @@
+/*
+* Copyright (c) 2024 RIT Launch Initiative
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef C_TENANT_H
+#define C_TENANT_H
+
+class CTenant {
+public:
+    /**
+     * Constructor
+     * @param name Name of the tenant
+     */
+    explicit CTenant(const char *name);
+
+    /**
+     * Destructor
+     */
+    virtual ~CTenant() = default;
+
+    /**
+     * Initialize the tenant
+     */
+    virtual void Startup() {};
+
+    /**
+     * Perform any other initializations after all tenants in the same task have been initialized
+     */
+    virtual void PostStartup() {};
+
+    /**
+     * Run the tenant
+     */
+    virtual void Run() = 0;
+protected:
+    const char *name;
```

Comment:
Debating on using std::string since this CTenant is created on init so it might be fine to do use heap here. Dont think theres an issue using a normal c string here tho